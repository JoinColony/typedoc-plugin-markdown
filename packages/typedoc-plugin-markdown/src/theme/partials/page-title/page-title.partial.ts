import { PageEvent, ParameterReflection } from 'typedoc';
import { MarkdownThemeRenderContext } from '../../theme.context';
import { escapeChars } from '../../theme.utils';

export const pageTitlePartial = (
  context: MarkdownThemeRenderContext,
  props: PageEvent<any>,
  shouldEscape = true,
) => {
  const md: string[] = [''];
  if (
    props.model &&
    props.model.kindString &&
    props.url !== props.project.url
  ) {
    md.push(`${props.model.kindString} `);
  }
  if (props.url === props.project.url) {
    md.push(props.project.name);
  } else {
    md.push(shouldEscape ? escapeChars(props.model.name) : props.model.name);
    if (props.model.typeParameters) {
      const typeParameters = props.model.typeParameters
        .map((typeParameter: ParameterReflection) => typeParameter.name)
        .join(', ');
      md.push(`<${typeParameters}${shouldEscape ? '\\>' : '>'}`);
    }
  }
  return md.join('');
};
