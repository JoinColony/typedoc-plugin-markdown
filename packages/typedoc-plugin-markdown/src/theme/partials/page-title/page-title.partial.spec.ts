import { ProjectReflection } from 'typedoc';
import { pageTitlePartial } from './page-title.partial';

describe(`Page Title Partial:`, () => {
  let project: ProjectReflection;

  beforeAll(async () => {
    project = global.getProject('generics.ts');
  });

  test(`should compile title with type params'`, () => {
    expect(
      pageTitlePartial(global.getMockContext(), {
        project: { url: 'README.md' },
        model: project.findReflectionByName('ClassWithTypeParams'),
      } as any),
    ).toMatchSnapshot();
  });
});
