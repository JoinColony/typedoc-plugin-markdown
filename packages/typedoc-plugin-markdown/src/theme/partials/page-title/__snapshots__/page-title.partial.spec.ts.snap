// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Title Partial: should compile title with type params' 1`] = `"Class: ClassWithTypeParams<T, V\\\\>"`;
